overlord/ifacestate/helpers.go

```diff
@@ -153,6 +156,38 @@ func (m *InterfaceManager) regenerateAllSecurityProfiles() error {
 	return nil
 }
 
+// renameCorePlugConnections renames connections from plugs to slots on the
+// core snap if the plug and slot name are the same the plug gets suffixed with
+// "-plug". This matches a similar rename in addSnaps above.
+func (m *InterfaceManager) renameCorePlugConnections() error {
+	conns, err := getConns(m.state)
+	if err != nil {
+		return err
+	}
+	changed := false
+	for _, plugName := range []string{"network-bind", "core-support"} {
+		// old connection, note that slotRef is the same in both
+		slotRef := interfaces.SlotRef{Snap: "core", Name: plugName}
+		oldPlugRef := interfaces.PlugRef{Snap: "core", Name: plugName}
+		oldConnRef := interfaces.ConnRef{PlugRef: oldPlugRef, SlotRef: slotRef}
+		oldID := oldConnRef.ID()
+		// new connection
```

Comment:
(FTR I think it's alright, if a little surprising, to have it this way instead of inside the if; don't change it if you'd only start the tests over for this change)